specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,981 @@
+      "description": "Describes a managed rule definition.",
+      "properties": {
+        "ruleId": {
+          "description": "Identifier for the managed rule.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "defaultState": {
+          "description": "Describes the default state for the managed rule.",
+          "readOnly": true,
+          "$ref": "#/definitions/ManagedRuleEnabledState"
+        },
+        "defaultAction": {
+          "description": "Describes the default action to be applied when the managed rule matches.",
+          "readOnly": true,
+          "$ref": "#/definitions/ActionType"
+        },
+        "description": {
+          "description": "Describes the functionality of the managed rule.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "ManagedRuleExclusion": {
+      "required": [
+        "matchVariable",
+        "selectorMatchOperator",
+        "selector"
+      ],
+      "description": "Exclude variables from managed rule evaluation.",
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "enum": [
+            "RequestHeaderNames",
+            "RequestCookieNames",
+            "QueryStringArgNames",
+            "RequestBodyPostArgNames",
+            "RequestBodyJsonArgNames"
+          ],
+          "description": "The variable type to be excluded.",
+          "x-ms-enum": {
+            "name": "ManagedRuleExclusionMatchVariable",
+            "modelAsString": true
+          }
+        },
+        "selectorMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith",
+            "EqualsAny"
+          ],
+          "description": "Comparison operator to apply to the selector when specifying which elements in the collection this exclusion applies to.",
+          "x-ms-enum": {
+            "name": "ManagedRuleExclusionSelectorMatchOperator",
+            "modelAsString": true
+          }
+        },
+        "selector": {
+          "type": "string",
+          "description": "Selector value for which elements in the collection this exclusion applies to."
+        }
+      }
+    },
+    "ActionType": {
+      "description": "Defines the action to take on rule match.",
+      "type": "string",
+      "enum": [
+        "Allow",
+        "Block",
+        "Log",
+        "Redirect"
+      ],
+      "x-ms-enum": {
+        "name": "ActionType",
+        "modelAsString": true
+      }
+    },
+    "ManagedRuleSetActionType": {
+      "description": "Defines the action to take when a managed rule set score threshold is met.",
+      "type": "string",
+      "enum": [
+        "Block",
+        "Log",
+        "Redirect"
+      ],
+      "x-ms-enum": {
+        "name": "ManagedRuleSetActionType",
+        "modelAsString": true
+      }
+    },
+    "ManagedRuleEnabledState": {
```

Comment:
>ManagedRuleEnabledState [](http://example.com/codeflow?start=5&length=23)

ManagedRuleState?

Reply:
Agree with the name, is a nit but good to update